### DISCUSSION TITLE
COCO-stuff for DetectoRS

### DISCUSSION THREAD

**Discussion by enemni** [open-mmlab/mmdetection#9230] [Q&A]
\Hello, 

when running ```python mmdetection/tools/misc/browse_dataset.py``` I got this error 

```
mmdetection/mmdet/datasets/pipelines/loading.py", line 372, in _load_semantic_seg
    filename = osp.join(results['seg_prefix'],
  File "/usr/lib/python3.8/posixpath.py", line 76, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

In my config file I uncommented   `seg_prefix='data/coco/stuffthingmaps/train2017` since I was not able to generate a COCO-stuff version of my data. Is the COCO-stuff for the DetectoRS required? If yes as explained in the documentation, how can I create it for my own dataset?

thanks a lot,

**Comment by TaWeiYeh** [comment]
I figured out how to make COCO-stuff data. Take a look at the thing and stuff classes [here](https://github.com/nightrome/cocostuff/blob/master/labels.md). 
The index of that class is the RGB values. However, remember to abstract all indexes by one. 
For example, the class of person has index 1. The value of a person class is 0. 

(Update) The output stuff and things image should be a **2D image ONLY**. That means we are saving a 2D array to an image. Unfortunately, Matplotlib could not save a 2D image. I use PIL package instead. 
Here is my code, 
`output_img = np.zeros((height, width), dtype=np.int32)`
`result = Image.fromarray(output_img.astype(np.uint8))`
`result.save(seg_filepath)`

![image](https://user-images.githubusercontent.com/15456906/200189767-db48bb9d-f26e-49f2-9bca-6d0da6d30c8f.png)
